### DISCUSSION TITLE
Problem with @tailwind base in own .css (Unknown at rule @tailwindcss(unknownAtRules)

### DISCUSSION THREAD

**Discussion by mxcdh** [tailwindlabs/tailwindcss#8857] [Help]
I have a problem with custom css in tailwind and vite. 
My maian.js look like:
```
import "./styles.css";
// import "tailwindcss/tailwind.css";

document.querySelector("#app").innerHTML = `
  <h1>Hello Vite!</h1>
  <a href="https://vitejs.dev/guide/features.html" target="_blank">Documentation</a>
`;

```

My styles.css looks like:
```
@tailwind base;
@tailwind components;
@tailwind utilities;

.header-nav-section {
  @apply bg-gradient-to-br from-gray-100e to-gray-300;
}
```

But vscode informed me that:
[![enter image description here][1]][1]

What is wrong, I readed documentation and this case will working. 

P.S
Full repo:
https://github.com/mxcdh/vite-tailwind

  [1]: https://i.stack.imgur.com/9LnbS.png

**Comment by arisac** [comment]
vscode doesn't know `@tailwind` and `@apply`
try use Tailwind CSS IntelliSense https://marketplace.visualstudio.com/items?itemName=bradlc.vscode-tailwindcss

**Reply by mxcdh** [reply]
I have Tailwind CSS IntelliSense.
When I uncommited: // import "tailwindcss/tailwind.css";
it's working, but when I commited this line and add: import "./styles.css"; Not working.

In next.js is working.

**Reply by arisac** [reply]
The problem may differ depends Each person's VS Code configuration and installed extensions. Especially the lint feature, other css helper or lint extensions may also cause problems.
You can add files.associations in your workspace: `.vscode/settings.json`, it's recommended on the IntelliSense page.
```json
{
  "files.associations": {
    "*.css": "tailwindcss"
  }
}
```
I have different **vscode settings** for different project, which differs for different project needs, and some can be complex. I know it's extra work, but it keeps everyone on have the same vscode settings for each repo whatever their local vscode workspace settings like.

**BTW**: I have [PostCSS Language Support extension](https://marketplace.visualstudio.com/items?itemName=csstools.postcss) installed, which works well with tailwind's IntelliSense. And most of the time I don't need to create extra settings for tailwind.

**Reply by LameckMeshack** [reply]
> ```json
> "*.css": "tailwindcss"
> ```

thank you

**Reply by DamianRivas** [reply]
@arisac thank you so much for sharing this. I wish this was easily spottable in documentation
